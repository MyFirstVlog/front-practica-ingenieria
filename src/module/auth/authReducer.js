import { types } from "../../constants/types";

export const authReducer = (state, action) => {
	switch (action.type) {
		case types.login:
			return {
				...state,
				logged: true,
				email: action.payload.email,
			};

		case types.logout:
			return {
				...state,
				logged: false,
				email: null,
			};

		default:
			return state;
	}
};
