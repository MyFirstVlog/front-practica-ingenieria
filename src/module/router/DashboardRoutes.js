import React from "react";
import { Redirect, Switch, Route } from "react-router-dom";
import { NavBar } from "../nav/NavBar";
import { BookingsPage } from "../pages/BookingsPage";
import { GestionPage } from "../pages/GestionPage";
import { SchedulingPage } from "../pages/SchedulingPage";
import { StatsPage } from "../pages/StatsPage";

export const DashboardRoutes = () => {
	return (
		<>
			<NavBar />

			<div className="mt-2">
				<Switch>
					<Route exact path="/" component={GestionPage} />
					<Route exact path="/scheduling" component={SchedulingPage} />
					<Route exact path="/bookings" component={BookingsPage} />
					<Route exact path="/stats" component={StatsPage} />

					<Redirect to="/" />
				</Switch>
			</div>
		</>
	);
};
