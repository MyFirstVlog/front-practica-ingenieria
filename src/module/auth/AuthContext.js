import React from "react";
import { createContext, useReducer } from "react";
import { authReducer } from "./authReducer";

export const AuthContext = createContext();

const initialState = {
	logged: false,
	email: "",
};

export const AuthProvider = ({ children }) => {
	const [authState, dispatch] = useReducer(authReducer, initialState);

	return (
		<AuthContext.Provider
			value={{
				authState,
				dispatch,
			}}
		>
			{children}
		</AuthContext.Provider>
	);
};
