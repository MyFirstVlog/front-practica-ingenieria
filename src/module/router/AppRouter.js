import React, { useContext } from "react";
import { BrowserRouter as Router, Switch } from "react-router-dom";
import { LoginPage } from "../pages/LoginPage";
import { PublicRoute } from "./PublicRoutes";
import { AuthContext } from "../auth/AuthContext";
import { PrivateRoute } from "./PrivateRoutes";
import { DashboardRoutes } from "./DashboardRoutes";

export const AppRouter = () => {
	const { authState } = useContext(AuthContext);

	return (
		<Router>
			<div>
				<Switch>
					<PublicRoute
						isAuthenticated={authState.logged}
						path="/login"
						component={LoginPage}
					/>
					<PrivateRoute
						isAuthenticated={authState.logged}
						path="/"
						component={DashboardRoutes}
					/>
				</Switch>
			</div>
		</Router>
	);
};
