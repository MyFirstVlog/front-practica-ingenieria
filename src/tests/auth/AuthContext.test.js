import { renderHook, act } from "@testing-library/react-hooks";
import React from "react";
import { mount } from "enzyme";
import { useReducer } from "react";
import { AuthContext, AuthProvider } from "../../module/auth/AuthContext";
import { authReducer } from "../../module/auth/authReducer";
import { AppRouter } from "../../module/router/AppRouter";
import { initialState } from "./authReducerMockConstants";

describe("should rest AuthProvider", () => {
	test("should test useReducer hook and go to public routes", () => {
		const { result } = renderHook(() =>
			useReducer(authReducer, initialState.initInput)
		);

		const [, dispatch] = result.current;

		act(() => {
			dispatch(initialState.actionLogout);
		});

		const [authState] = result.current;

		const wrapper = mount(
			<AuthContext.Provider value={{ authState }}>
				<AppRouter />
			</AuthContext.Provider>
		);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("label").at(0).text()).toBe("Email");
		expect(wrapper.find("label").at(1).text()).toBe("ContraseÃ±a");
	});

	test("should test useReducer hook and go to dashboard", () => {
		const { result } = renderHook(() =>
			useReducer(authReducer, initialState.initInput)
		);

		const [, dispatch] = result.current;

		act(() => {
			dispatch(initialState.actionLogin);
		});

		const [authState] = result.current;

		const wrapper = mount(
			<AuthContext.Provider value={{ authState }}>
				<AppRouter />
			</AuthContext.Provider>
		);

		expect(wrapper).toBeDefined();
	});

	test("should test useReducer hook and check auth provider", () => {
		const { result } = renderHook(() =>
			useReducer(authReducer, initialState.initInput)
		);

		const [, dispatch] = result.current;

		act(() => {
			dispatch(initialState.actionLogin);
		});

		const wrapper = mount(
			<AuthProvider>
				<AppRouter />
			</AuthProvider>
		);

		expect(wrapper).toBeDefined();
	});
});
