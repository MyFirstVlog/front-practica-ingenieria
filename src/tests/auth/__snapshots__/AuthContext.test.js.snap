// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rest AuthProvider should test useReducer hook and go to public routes 1`] = `
<div>
  <div
    className="mt-5 card form-container card"
  >
    <div
      className="form-header text-center"
    >
      <img
        alt="cidenet"
        className="img-cidenet"
        src="https://res.cloudinary.com/crunchbase-production/image/upload/c_lpad,h_256,w_256,f_auto,q_auto:eco,dpr_1/ltgilmylvmrd2mnrsjem"
      />
    </div>
    <div
      className="card-body"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="mb-4"
        >
          <label
            className="form-label"
            htmlFor="exampleInputEmail1"
          >
            Email
          </label>
          <input
            aria-describedby="emailHelp"
            className="form-control"
            id="exampleInputEmail1"
            name="email"
            onChange={[Function]}
            type="email"
          />
        </div>
        <div
          className="mb-4"
        >
          <label
            className="form-label"
            htmlFor="exampleInputPassword1"
          >
            Contrase√±a
          </label>
          <input
            className="form-control"
            id="exampleInputPassword1"
            name="password"
            onChange={[Function]}
            type="password"
          />
        </div>
        <div
          className="text-center"
        >
          <button
            className="btn btn-primary mb-2"
            disabled={true}
            type="submit"
          >
            <span
              style={
                Object {
                  "marginRight": "7px",
                }
              }
            >
              <i
                className="fa-solid fa-arrow-right-to-bracket"
              />
            </span>
            Ingresar
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
